src/main/java/seedu/address/wrapper/UserInfoRoute.java

```diff
@@ -0,0 +1,103 @@
+import seedu.address.wrapper.exceptions.ResponseParseException;
+
+/**
+ * Class representing routes taken to get user information
+ */
+public final class UserInfoRoute {
+    //@@author arnav-ag
+
+    public static final String BASE_GITHUB_URL = "https://api.github.com";
+    private static final String SERVER_DOWN_STATUS_MESSAGE = "Unable to get information from API.";
+
+    private static final String GET_USER_BASE_PATH = "/users/";
+    private final String path;
+    private final Storage storage;
+    private final String username;
+
+    private UserInfoRoute(String path, String username, Storage storage) {
+        requireAllNonNull(path, username, storage);
+        this.path = path;
+        this.storage = storage;
+        this.username = username;
+    }
+
+    public static UserInfoRoute getUserInfoRoute(String username, Storage storage) {
+        requireAllNonNull(username);
+        return new UserInfoRoute(GET_USER_BASE_PATH + username, username, storage);
+    }
+
+    /**
+     * @param unirest Unirest instance to be used to carry out requests
+     * @return Request class pertaining to user information from GitHub
+     */
+    public UserInfoRequest createRequest(UnirestInstance unirest) {
+        requireAllNonNull(unirest);
+
+        return new UserInfoRequest(unirest, BASE_GITHUB_URL + this.path, storage, username);
+    }
+
+    public String getPath() {
+        return this.path;
+    }
+
+    /**
+     * Class representing request needed to retrieve user information from GitHub
+     */
+    public static class UserInfoRequest {
+        private final Storage storage;
+
+        private final UnirestInstance unirest;
+        private final String url;
+
+        private final String username;
+
+        UserInfoRequest(UnirestInstance unirest, String url, Storage storage, String username) {
+            requireAllNonNull(unirest, url, storage, username);
+            this.unirest = unirest;
+            this.url = url;
+            this.storage = storage;
+            this.username = username;
+        }
+
+        public JSONObject getJson() {
+            var response = this.unirest.get(this.url).asString().getBody();
+
+            try {
+                return new JSONObject(response);
+            } catch (JSONException e) {
+                throw new ResponseParseException("Unable to parse API result.");
+            }
+        }
+
+        /**
+         * @param fileUrl URL pertaining to the user's avatar image
+         */
+        public void downloadAvatarImage(String fileUrl) {
+            Unirest.get(fileUrl).thenConsume(rawResponse -> {
+                try {
+                    BufferedImage image = ImageIO.read(rawResponse.getContent());
+                    storage.saveImage(image, username + ".png");
+                } catch (IOException e) {
+                    throw new ResponseParseException("Error saving user avatar.");
+                }
+            });
+
+        }
```

Comment:
I think this part should be at `UserInfoWrapper` instead of `UserInfoRequest` because `UserInfoRequest` should not include or consist `downloadAvatarImage` function.

A better abstraction would be splitting request in `UserInfoRoute` class.

```java
public UserInfoRequest createUserInfoRequest(UnirestInstance unirest) {
    assert unirest != null;

    return new UserInfoRequest(unirest, BASE_GITHUB_URL + this.path, storage, username);
}

public UserInfoRequest createUserImageRequest(UnirestInstance unirest, String avatarUrl) {
    assert unirest != null;

    return new UserInfoRequest(unirest, BASE_GITHUB_URL + avatarUrl, storage, username);
}
```